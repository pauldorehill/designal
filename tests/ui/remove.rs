use designal::Designal;

#[derive(Designal)]
#[designal(trim_start = "Human")]
#[designal(remove)]
struct HumanBean();

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean1 {
    #[designal(remove)]
    #[designal(remove)]
    taste: String,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean2 {
    #[designal(remove, rename = "Yum")]
    taste: String,
}

fn main() {}
