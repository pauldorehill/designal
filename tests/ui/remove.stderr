error: Remove is not valid at the container level
 --> $DIR/remove.rs:5:12
  |
5 | #[designal(remove)]
  |            ^^^^^^

error: You should only `remove` once
  --> $DIR/remove.rs:12:16
   |
12 |     #[designal(remove)]
   |                ^^^^^^

error: You have removed and renamed a field
  --> $DIR/remove.rs:19:16
   |
19 |     #[designal(remove, rename = "Yum")]
   |                ^^^^^^
